src/pudl/transform/eia861.py

```diff
@@ -920,6 +919,49 @@ def dynamic_pricing(tfr_dfs):
         dict: A dictionary of transformed EIA 861 dataframes, keyed by table name.
 
     """
+    idx_cols = [
+        "utility_id_eia",
+        "state",
+        "balancing_authority_code_eia",
+        "report_date",
+    ]
+
+    class_attributes = [
+        'critical_peak_pricing',
+        'critical_peak_rebate',
+        'real_time_pricing_program',
+        'time_of_use_pricing_program',
+        'variable_peak_pricing_program'
+    ]
+
+    raw_dp = tfr_dfs["dynamic_pricing_eia861"].copy()
+
+    ###########################################################################
+    # Tidy Data:
+    ###########################################################################
+
+    logger.info("Tidying the EIA 861 Dynamic Pricing table.")
+    tidy_dp = _tidy_customer_class_dfs(raw_dp, idx_cols)
+
+    ###########################################################################
+    # Transform Values:
+    # * Make Y/N's into booleans and X values into pd.NA
+    ###########################################################################
+
+    logger.info(
+        "Performing value transformations on EIA 861 Dynamic Pricing table.")
+    for col in class_attributes:
+        tidy_dp[col] = (
+            tidy_dp[col].replace({'Y': True, 'N': False})
+            .apply(lambda x: x if x in [True, False] else pd.NA)
+        )
+    transformed_dp = tidy_dp
```

Comment:
Is this the line you were asking about on Slack? It looks like this is a no-op, since you clobber the value of `transformed_dp` immediately below. It would probably make sense to just remove this line entirely.

Reply:
yeah -- the only reason I included it was for the sake of the name. I couldn't figure out how to elegantly name the prior portion `transformed_dp` as `tidy_dp` wasn't in keeping with our prior naming practice. If you think it's fine to keep the name `tidy_dp` (or if you know a way around that) then I'll get rid of that superfluous line.

Reply:
I think it's fine to nix the line -- the naming is just so that when we go back and re-organize this into different steps we the humans know where the boundaries are.